optax/_src/alias.py

```diff
@@ -56,32 +57,85 @@ def adagrad(
   )
 
 
-def adam(learning_rate: ScalarOrSchedule,
-         b1: float = 0.9,
-         b2: float = 0.999,
-         eps: float = 1e-8,
-         eps_root: float = 0.0) -> GradientTransformation:
+def adam(
+    learning_rate: ScalarOrSchedule,
+    b1: float = 0.9,
+    b2: float = 0.999,
+    eps: float = 1e-8,
+    eps_root: float = 0.0) -> GradientTransformation:
+  """The classic Adam optimiser.
+
+  Adam is an SGD variant with learning rate adaptation. The `learning_rate`
+  used for each weight is computed from estimates of first and second order
+  moments of the gradients (using suitable exponential moving averages).
+
+  References:
+    [Kingma et al, 2014](https://arxiv.org/abs/1412.6980)
+
+  Args:
+    learning_rate: this is a fixed global scaling factor.
+    b1: the decay used to track the first moment of past gradients.
+    b2: the decay used to track the second moment of past gradients.
+    eps: a small constant applied to denominator outside of the square root
+      (as in the Adam paper) to avoid dividing by zero when rescaling.
+    eps_root: (default `0`), a small constant applied to denominator inside the
+      square root (as in RMSProp), to avoid dividing by zero when rescaling.
```

Comment:
Suggestion: (cleaner this way?)

```suggestion
    eps_root: a small constant applied to denominator inside the
      square root (as in RMSProp), to avoid dividing by zero when rescaling (default: 0).
```

Also, maybe we should use "RMSProp" instead of "RmsProp"? WDYT

Reply:
Agreed, updated to use RMSProp